### DISCUSSION TITLE
when location to other html, the go context is gone?

### DISCUSSION THREAD

**Discussion by askie** [wailsapp/wails#1512] [General]
wails : v2
my code in frontend/index.html
```
<script>
        location.href = "./src/test.html";
    </script>
```
code in src/test.html
```
Greek('aaa').then(r=>{

})
```
error:
```
Greek is undefined......
```
why in wails window,try location.href to other html or url 
, the go Context function is gone?

how can I resove?

**Comment by leaanthony** [comment]
Yes, if you navigate away you will lose context.

**Comment by daodao97** [comment]
you can add the script in your `src/test.html`
```html
<head>
    <script src="/wails/ipc.js"></script>
    <script src="/wails/runtime.js"></script>
</head>
```

then call wails method

```html
    <script>
        runtime.LogError('a custom page log')
        window.go.app.App.Xxxxx().then(res => {
           console.log(res)
        })
    </script>
```

**Reply by xieting1990** [reply]
I tested according to this method, but I didn't get what I wanted
<img width="615" alt="image" src="https://user-images.githubusercontent.com/34619801/188635480-c8c2e6a8-bdf2-4782-ab4c-07a0744b8052.png">

**Reply by termlabx** [reply]
> ```
>  <script src="/wails/ipc.js"></script>
>     <script src="/wails/runtime.js"></script>
> ```

This should be in the wails document

**Comment by misitebao** [comment]
https://wails.io/docs/next/guides/frontend/
